pipeline/tests/asm.rs

```diff
@@ -790,3 +791,34 @@ fn trait_parsing() {
     let f = "asm/trait_parsing.asm";
     verify_asm(f, Default::default());
 }
+
+#[test]
+fn dynamic_fixed_cols() {
+    let f = "asm/dynamic_fixed_cols.asm";
+    let i = [];
+    verify_asm(f, slice_to_vec(&i));
+}
+
+#[test]
+fn types_in_expressions() {
+    let input = r#"
+        enum O<X> {
+            A(X),
+            B,
+        }
+        let f: -> Constr = || {
+            let g: expr[] = [1, 2];
+            let h: expr -> O<expr> = |i| O::A::<expr>(i);
+            match h(g[1]) {
+                O::A(x) => x,
+            } = 0
+        };
+        machine Main {
+            col witness w;
+            f();
+        }
+        "#;
+    let output = asm_string_to_pil::<GoldilocksField>(input).to_string();
+    let expected = "    2 = 0;\n";
+    assert_eq!(output, expected);
```

Comment:
Maybe this test should be moved to one of the compiler crates instead of Pipeline which does end-to-end stuff?

Reply:
We don't have a crate that spans asm and pil, so pipeline is the correct crate IMO